app/src/main/java/com/x8bit/bitwarden/data/platform/manager/ciphermatching/CipherMatchingManagerImpl.kt

```diff
@@ -210,9 +212,15 @@ private fun LoginUriView.checkForMatch(
             UriMatchType.EXACT -> exactIfTrue(loginViewUri == matchUri)
 
             UriMatchType.HOST -> {
-                val loginUriHost = loginViewUri.getHostWithPortOrNull()
-                val matchUriHost = matchUri.getHostWithPortOrNull()
-                exactIfTrue(matchUriHost != null && loginUriHost == matchUriHost)
+                if (loginViewUri.hasPort() && matchUri.hasPort()) {
```

Comment:
I think we could technically just check if the `matchUri` has a port since it is the value that might not have it, whereas the `loginViewUri` is the value stored to disk and _should_ be accurate.

That could make the functionality more confusing but it is an option.

Reply:
That's what I had initially too in my snippet. Assuming this may change under our feet at some point in the future, I thought it could start failing to match cases where users just set a domain / IP with the knowledge that it didn't matter back then, and it would be difficult for us to troubleshoot why. 🤔

Reply:
I would hope that it would not change under our feet since the AutofillService provides us the `webDomain` and `webScheme` as 2 separate components and we have to put it together ourselves. Hopefully the port would be added as a new property that we would have to manage ourselves. Having said that, it could happen. 

Do you think I should change this to only check for the `matchUri.hasPort()`

Reply:
I would leave it as is.